makefile

```diff
@@ -108,7 +108,11 @@ endif
 STAN_FLAGS=$(STAN_FLAG_THREADS)$(STAN_FLAG_MPI)$(STAN_FLAG_OPENCL)
 
 ifeq ($(OS),Windows_NT)
+ifeq ($(shell expr $(CXX_MAJOR) \>= 8), 1)
```

Comment:
Should we also be checking they are using gcc and not clang or would it be fine in that case as well?

Reply:
Clang on windows with cmdstandoes not work at all (https://github.com/stan-dev/cmdstan/issues/736) so that didnt cross my mind. Will add a note there if we ever figure that out. 

Other option would be to keep the precompiled headers off in the off chance someone tries this with clang on Windows?

Reply:
Changed the logic to also handle clang on Windows. As we have no idea if it works, I opted to disabling it by default.